samza-api/src/main/java/org/apache/samza/drain/DrainNotification.java

```diff
@@ -35,9 +35,23 @@ public class DrainNotification {
    */
   private final String runId;
 
-  public DrainNotification(UUID uuid, String runId) {
+  /***/
+  private final DrainMode drainMode;
+
+  public DrainNotification(UUID uuid, String runId, DrainMode drainMode) {
+    Preconditions.checkNotNull(uuid);
+    Preconditions.checkNotNull(runId);
+    Preconditions.checkNotNull(drainMode);
     this.uuid = uuid;
     this.runId = runId;
+    this.drainMode = drainMode;
+  }
+
+  /**
+   * Creates a DrainNotification in {@link DrainMode#SIMPLE} mode.
+   * */
+  public static DrainNotification simple(UUID uuid, String runId) {
```

Comment:
let's just call it create(). We can have another variant factory method for create with other mode.

Reply:
Done